### PR TITLE
Implement initial `FilecoinRecordInterpreter`

### LINKED ISSUE #15: Implement FilecoinRecordInterpreter
See https://github.com/ipfs-shipyard/w3rc/pull/14 once merged

Implement the FilecoinRecordInterpreter per instructions, move around and test as needed

### PR DESCRIPTION
**Pull request by masih**
Implement the initial `FilecoinRecordInterpreter` specified in #14.

Fixes #15

### CODE REVIEW

File: planning/fil_record_interpreter.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/filecoin-project/indexer-reference-provider/metadata"
+	"github.com/filecoin-project/storetheindex/api/v0"
+	"github.com/ipfs-shipyard/w3rc/contentrouting"
+)
+
+// ------- FORK START --------------
+// TODO: Remove forked code below once #14 is merged.
+// Forked to unblock implementation.
+// See: https://github.com/ipfs-shipyard/w3rc/pull/14
+type (
```
Comment (masih) [inline planning/fil_record_interpreter.go:14]: Pending #14 merge to `main`. Forking code for now to unblock work and reduce merge conflicts

File: planning/fil_record_interpreter.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	fv1d, err := metadata.DecodeFilecoinV1Data(dtm)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: How should policies argument impact the returned results?
```
Comment (masih) [inline planning/fil_record_interpreter.go:71]: @hannahhoward  I would love your help in answering this question please.

Comment (hannahhoward) [reply planning/fil_record_interpreter.go:72]: I agree. This is weird maybe we remove the policy list here?

**Comment by masih** [comment]
@hannahhoward a quick heads-up: now that #14 is merged I have reflected the changes in my PR and removed forked code. Thanks

File: planning/fil_record_interpreter.go (hunk 2 above)
Comment (hannahhoward) [reply planning/fil_record_interpreter.go]: I am inclined to merge as is for now given that we are still developing code here.

**Review by hannahhoward — APPROVED** [review]
LGTM

**Pull request merged**